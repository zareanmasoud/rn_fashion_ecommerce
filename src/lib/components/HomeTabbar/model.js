// @flow

export type Buttons = {
  press: Function,
  src: string,
  text: string,
};
