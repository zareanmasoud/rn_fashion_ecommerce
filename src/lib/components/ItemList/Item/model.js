// @flow

export type ItemModel = {
  id: string,
  photo: string,
};
