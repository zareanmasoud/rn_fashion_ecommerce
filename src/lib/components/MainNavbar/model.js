// @flow

export type Button = {
  action: Function,
  src: string,
};
