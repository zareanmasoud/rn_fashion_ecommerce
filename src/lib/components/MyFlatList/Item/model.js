// @flow

export type Data = {
  id: string,
  photo: string,
};
