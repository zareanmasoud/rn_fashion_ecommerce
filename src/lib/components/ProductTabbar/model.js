// @flow

export type Specs = {
  action: Function,
  src: string,
  text: string,
};
