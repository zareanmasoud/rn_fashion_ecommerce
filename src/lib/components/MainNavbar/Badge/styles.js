import {StyleSheet} from 'react-native';

const styles = StyleSheet.create({
  container: {
    position: 'absolute',
    top: 0,
    right: 0,
  },
});

export default styles;
