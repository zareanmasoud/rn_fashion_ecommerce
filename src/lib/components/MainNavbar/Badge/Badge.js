import React from 'react';
import {View, Text} from 'react-native';
import styles from './styles';

type Props = {
  cart: any,
  cartTextTestID: string,
};

export default function Badge({cart, cartTextTestID}: Props) {
  return (
    <View style={styles.container}>
      <Text testID={cartTextTestID}>{cart}</Text>
    </View>
  );
}
