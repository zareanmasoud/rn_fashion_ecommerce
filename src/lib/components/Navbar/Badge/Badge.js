import React from 'react';
import {View, Text} from 'react-native';
// eslint-disable-next-line import/no-extraneous-dependencies
import styles from './styles';

type Props = {
  resources: any,
};

export default function Badge({resources}: Props) {
  const {cart, cartTextTestID} = resources;

  return (
    <View style={styles.container}>
      <Text testID={cartTextTestID}>{cart}</Text>
    </View>
  );
}
