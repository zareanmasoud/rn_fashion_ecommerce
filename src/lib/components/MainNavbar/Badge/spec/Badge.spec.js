import React from 'react';
import {shallow} from 'enzyme';
import Badge from '../Badge';
import model from './model';

describe('Badge Component', () => {
  describe('Rendering', () => {
    it('should match to snapshot', () => {
      const wrapper = shallow(
        <Badge cart={model.cart} cartTextTestID={model.cartTextTestID} />,
      );
      expect(wrapper).toMatchSnapshot();
    });
  });
});
