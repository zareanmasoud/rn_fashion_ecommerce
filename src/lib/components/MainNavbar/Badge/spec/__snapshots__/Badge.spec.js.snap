// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Badge Component Rendering should match to snapshot 1`] = `
<View
  style={
    Object {
      "position": "absolute",
      "right": 0,
      "top": 0,
    }
  }
>
  <Text />
</View>
`;
